### PR TITLE
Consolidate AccountCreator and SetupPortalAccountJob into RegisterUserAccount

### PR DESCRIPTION
**Pull request by sadleb**
### Description
When a user registers a new account using a link like:
https://platformweb/users/sign_up?u=0031J00001y1vQG

it's responsible for creating a new Canvas user account, enrolling the
user in the proper Canvas section, updating Salesforce to match,
and creating the local User record.

This logic was very hard to follow and there was a lot of obsolete stuff
in there for Highlander like DocuSign and syncing to the Join server.
This cleans up this flow to make it easier to understand in preparation
for changing the Sync part of this to leave them in the CohortSchedule
section when running the logic to adjust their enrollment when changes happen.

**TODO:** write specs for more graceful error handling

### Screenshots (if appropriate):

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/services/register_user_account.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+# Fully registers and creates a new User account in all of our systems
+# (aka Platform, Canvas and Salesforce) so that it's ready for use.
+# Everything must be consistent and in sync so the systems can work together.
+class RegisterUserAccount
```
Comment (sadleb) [inline app/services/register_user_account.rb:6]: This does the same thing that the deleted files `account_creator.rb`, `setup_portal_account.rb`, and `setup_portal_account_job.rb ` did. Is it easier to read and understand this way?

File: app/services/register_user_account.rb [hunk 2]
```diff
@@ -14,13 +14,25 @@ def initialize(sign_up_params)
...
 
   def run
     Honeycomb.start_span(name: 'RegisterUserAccount.run') do |span|
       span.add_field('salesforce.contact.id', @salesforce_participant.contact_id)
 
+      # Need to have the User record saved before the sync runs since it relies on it.
+      @new_user.skip_confirmation_notification!
+      span.add_field('user.id.created', @new_user.id)
+
+      # Do this before syncing their enrollment b/c if it fails we want to rollback.
```
Comment (sadleb) [inline app/services/register_user_account.rb:31]: Remove rollback language.

**Comment by rshipp** [comment]
bypassing pr approval based on standup discussion
